src/main/routes/receiver.ts

```
@@ -1,49 +1,192 @@
-    if (req.query.jwt) {
-      const sessionCookie = config.get<string>('session.cookieName')
-      new Cookies(req, res).set(sessionCookie, req.query.jwt, { sameSite: 'lax' })
-    }
-
-    const user: User = res.locals.user
-    const atLeastOneClaimIssued: boolean = (await ClaimStoreClient.retrieveByClaimantId(user.id)).length > 0
-    const claimAgainstDefendant = await ClaimStoreClient.retrieveByDefendantId(user.id)
-    const atLeastOneResponse: boolean = claimAgainstDefendant.length > 0 &&
-      claimAgainstDefendant.some((claim: Claim) => !!claim.response)
-
-    if (atLeastOneClaimIssued || atLeastOneResponse) {
-      return res.redirect(DashboardPaths.dashboardPage.uri)
-    }
-    const draftClaimSaved: boolean = user.claimDraft.document && user.claimDraft.id !== undefined
-    const claimIssuedButNoResponse: boolean = (claimAgainstDefendant).length > 0
-      && !atLeastOneResponse
-
-    if (claimIssuedButNoResponse && draftClaimSaved) {
-      return res.redirect(DashboardPaths.dashboardPage.uri)
-    }
-
-    if (draftClaimSaved) {
-      return res.redirect(ClaimPaths.taskListPage.uri)
-    }
-
-    if (claimIssuedButNoResponse) {
-      return res.redirect(ResponsePaths.taskListPage
-        .evaluateUri({ externalId: claimAgainstDefendant.pop().externalId }))
-    }
-
-    return res.redirect(ClaimPaths.startPage.uri)
-  }))
+  .get(AppPaths.receiver.uri,
+    ErrorHandling.apply(async (req: express.Request,
+                               res: express.Response,
+                               next: express.NextFunction): Promise<void> => {
+      const cookies = new Cookies(req, res)
+
+      const authenticationToken = await getAuthToken(req)
+      if (authenticationToken) {
+        const user = await IdamClient
+          .retrieveUserFor(authenticationToken)
+          .catch(err => loginErrorHandler(req, res, next, err))
+        if (!user) {
+          return
```

Would be more clearer if we'd throw an error here instead of just returning. Or is there a reason for just returning?

---

I think 101-112 could be extracted and reused in the subsequent get, they look very similar.

---

it returns because the catch has already handled it.
was wondering about a try catch for that but it would be a bit yuck

---

have improved